### PR TITLE
Updated tr0n's Quest Board

### PR DESCRIPTION
**Pull request by sigtus**
* **Addressed Issue(s)**: none.

* **Description of Pull Request**: 

- Updated to the last version of the author.
- Changed values to constants.
- Optimized loops by using cache of getarraysize.
- Changed the party handling (doesn't stop when a player reached his quest but wants to help party member who haven't finished the same quest).

### CODE REVIEW

**Comment by Lemongrass3110** [comment]
https://travis-ci.org/rathena/rathena/jobs/486100597#L1171

Missing string concatenation

File: npc/custom/quests/questboard.txt [hunk 1]
```diff
@@ -61,136 +63,150 @@ prontera,129,215,5	script	Questboard	857,{
...
+		for( set .@i,0; .@i < .collectionquestcount; set .@i,.@i+1){
+			if (.@i) set .@collectmenu$,.@collectmenu$+":";
+			set .@collectmenu$, .@collectmenu$ + "[" + getd("." +(.@i+1)+"_collectionmin") + " - " + getd("." +(.@i+1)+"_collectionmax") + "] " + getd("." + (.@i+1) + "_collectionname$");
+		}
+		set .@selection,select(.@collectmenu$);
+		if(.quest_repeat == true){
+			if(gettimetick(2) < getd(.@selection + "_collection_delay")){
+				set .@time_left, getd(.@selection + "_collection_delay")-gettimetick(2);
+				mes "[^FF7700Questboard^000000]";
+				mes "You have to wait ^0000FF"+Time2Str(.@time_left)^000000 to do this quest again.";
```
Comment (Lemongrass3110) [inline npc/custom/quests/questboard.txt]:
```suggestion
				mes "You have to wait ^0000FF"+Time2Str(.@time_left)+"^000000 to do this quest again.";
```

File: npc/custom/quests/questboard.txt [hunk 2]
```diff
@@ -225,29 +243,54 @@ L_checkmobs2:
...
-			if(getd(currentquest$+"_"+.@currentmob+"_"+.@i+"_killcount")>=getd("."+currentquest$+"_huntingmob["+.@i+"]")) end;
-			setd(currentquest$+"_"+.@currentmob+"_"+.@i+"_killcount", getd(currentquest$+"_"+.@currentmob+"_"+.@i+"_killcount")+1);
-			dispbottom getd("."+currentquest$+"_huntingname$")+": ["+strmobinfo(1,.@currentmob)+"] ("+ getd(currentquest$+"_"+.@currentmob+"_"+.@i+"_killcount")+"/"+getd("."+currentquest$+"_huntingmob["+.@i+"]")+")";
-			end;
+			if(getd(currentquest$+"_"+.@currentmob+"_"+.@i+"_killcount")<getd("."+currentquest$+"_huntingmob["+.@i+"]")){
+				setd(currentquest$+"_"+.@currentmob+"_"+.@i+"_killcount", getd(currentquest$+"_"+.@currentmob+"_"+.@i+"_killcount")+1);
+				dispbottom getd("."+currentquest$+"_huntingname$")+": ["+strmobinfo(1,.@currentmob)+"] ("+ getd(currentquest$+"_"+.@currentmob+"_"+.@i+"_killcount")+"/"+getd("."+currentquest$+"_huntingmob["+.@i+"]")+")";
+			}
+			if(getcharid(1) != 0 && .party_support == true){
+				getmapxy(.@map1$,.@x1,.@y1,UNITTYPE_PC);
```
Comment (aleos89) [inline npc/custom/quests/questboard.txt]: `UNITTYPE_PC` is now deprecated, and this parameter is now optional. You can now define it as `getmapxy(.@map1$,.@x1,.@y1);`

File: npc/custom/quests/questboard.txt [hunk 3]
```diff
@@ -225,29 +243,54 @@ L_checkmobs2:
...
+				set .@killerid, getcharid(3);
+				set .@currentquest$, currentquest$;
+				getpartymember getcharid(1),1;
+				getpartymember getcharid(1),2;
+				for(set .@j, 0; .@j < $@partymembercount; .@j++){
+					if(isloggedin($@partymemberaid[.@j], $@partymembercid[.@j])){
+						if(h_run==true && $@partymemberaid[.@j] != .@killerid && .@currentquest$ == getvar(currentquest$, $@partymembercid[.@j]) && readparam(HP, $@partymembercid[.@j]) > 0){
+							getmapxy(.@map2$,.@x2,.@y2,UNITTYPE_PC,rid2name($@partymemberaid[.@j]));
```
Comment (aleos89) [inline npc/custom/quests/questboard.txt]: Same here, just adjust `UNITTYPE_PC` to `BL_PC`.

**Comment by Atemo** [comment]
@sigtus Anything left?

**Comment by sigtus** [comment]
@Atemo Everything right.

**Comment by Atemo** [comment]
Thanks for contributing !

**Pull request merged**